google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -273,7 +275,8 @@ static UnitOfWorkType of(TransactionMode transactionMode) {
     this.autoPartitionMode = options.isAutoPartitionMode();
     this.maxPartitions = options.getMaxPartitions();
     this.maxPartitionedParallelism = options.getMaxPartitionedParallelism();
-    this.ddlClient = createDdlClient();
+    this.databaseAdminClient = spanner.createDatabaseAdminClient();
```

Comment:
This will create a separate admin client for each connection that is created, including underlying gRPC channels. That means that an application that uses a JDBC connection that opens 100 connections, all of a sudden uses 4 channels (for the normal Spanner instance) + 100 (or 400? I don't remember exactly what channel settings we are using for the generated admin clients) channels for the 100 DatabaseAdmin clients.

It also means that creating a new `Connection` is suddenly a much heavier operation, meaning that applications that do not use connection pooling could potentially experience a performance degradation.

Currently, creating a `Connection` instance is an non-blocking method, and creating multiple connections to the same database does not incur much cost (it only costs the bit of memory that is required to keep track of the `Connection` state, but it does not create any additional physical connections to Spanner).